src/connection.js

```diff
@@ -405,6 +422,62 @@ export default class Connection extends EventEmitter {
     return this.messageIo.on('secure', this.emit.bind(this, 'secure'));
   }
 
+  resolveIPaddresses(host, callback) {
+    let ipv4Addresses = undefined;
+    let ipv6Addresses = undefined;
+    //resolve IPV4
+    Dns.resolve4(host, (err, addresses) => {
+      ipv4Addresses = addresses || new Array();
+      if (ipv6Addresses) {
+        callback(ipv4Addresses.concat(ipv6Addresses));
+      }
+    });
+    
+    //resolve IPV6
+    Dns.resolve6(host, (err, addresses) => {
+      ipv6Addresses = addresses || new Array();
+      if (ipv4Addresses) {
+        callback(ipv4Addresses.concat(ipv6Addresses));
+      }
+    });
+  }
+ 
+  multiConnectOnPort(addresses, port, callback) {
```

Comment:
Mmh, I'm not sure. We can also leave it as-is for now to get this merged faster and then reiterate on it, if that makes sense.